api/src/main/java/edu/cornell/mannlib/vitro/webapp/web/directives/DataGetterDirective.java

```diff
@@ -0,0 +1,139 @@
+import freemarker.core.Environment;
+import freemarker.template.ObjectWrapper;
+import freemarker.template.TemplateDirectiveBody;
+import freemarker.template.TemplateException;
+import freemarker.template.TemplateModel;
+import freemarker.template.TemplateModelException;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.jena.ontology.OntModel;
+
+/**
+ * Freemarker directive to make substitutions in DataGetter and return the data
+ * in variable.
+ */
+public class DataGetterDirective extends BaseTemplateDirectiveModel {
+    private static final Log log = LogFactory.getLog(DataGetterDirective.class);
+
+    @Override
+    public void execute(Environment env, Map params, TemplateModel[] loopVars, TemplateDirectiveBody body)
+            throws TemplateException, IOException {
+        long startTime = System.nanoTime();
+        String dataGetterUri = getRequiredSimpleScalarParameter(params, "uri");
+        String variableName = getOptionalSimpleScalarParameter(params, "var");
+        HttpServletRequest req = (HttpServletRequest) env.getCustomAttribute("request");
+        VitroRequest vreq = new VitroRequest(req);
+        debug(getTimeSince(startTime) + "ms .1");
+        try {
+            debug(getTimeSince(startTime) + "ms .2");
+            OntModel model = vreq.getDisplayModel();
+            debug(getTimeSince(startTime) + "ms .3");
+            DataGetter dataGetter = DataGetterUtils.dataGetterForURI(vreq, model, dataGetterUri);
+            debug(getTimeSince(startTime) + "ms .4");
+            Map<String, Object> parameters = getOptionalHashModelParameter(params, "parameters");
+            debug(getTimeSince(startTime) + "ms .5");
+            applyDataGetter(dataGetter, env, parameters, variableName);
+            debug(getTimeSince(startTime) + "ms .6");
+        } catch (Exception e) {
+            handleException(dataGetterUri, "Could not process data getter '%s'", e);
+        }
+    }
+
+    /**
+     * Get the data from a DataGetter, provide variable values for substitution and
+     * store results in Freemarker environment variable.
+     * 
+     * @param dataGetter - DataGetter to execute
+     * @param env - Freemarker environment
+     * @param parameters - parameters to substitute in DataGetter
+     * @param overrideVariableName - name of Freemarker variable
+     * 
+     */
+    private static void applyDataGetter(DataGetter dataGetter, Environment env, Map<String, Object> parameters,
+            String overrideVariableName) throws TemplateModelException {
+        Map<String, Object> data = dataGetter.getData(parameters);
+        if (data != null) {
+            Object key = data.get("variableName");
+            if (key != null) {
+                Object value = data.get(key.toString());
+                setVariable(env, overrideVariableName, key.toString(), value);
```

Comment:
What do you think about suggestion below and simplification of setVariable method in this class (there will be only three parameters and if statement will be removed)
```suggestion
                setVariable(env, (!StringUtils.isBlank(overrideVariableName))?overrideVariableName:key.toString(), value);
```

Reply:
In my humble opinion proposed suggestion is less readable.
Extracted method serves the purpose to hide variable decision and assignment details.
Of course it could be inlined, but it seems to be that it is more readable in this form.
I updated method documentation, log messages and improved variable names.